cpp/src/arrow/record_batch.cc

```diff
@@ -360,9 +365,6 @@ Result<std::shared_ptr<Tensor>> RecordBatch::ToTensor(MemoryPool* pool) const {
     case Type::INT64:
       ConvertColumnsToTensor<Int64Type>(*this, result->mutable_data());
       break;
-    case Type::HALF_FLOAT:
-      ConvertColumnsToTensor<HalfFloatType>(*this, result->mutable_data());
-      break;
```

Comment:
You can keep this if we find that clearer, because the CType is defined for HalfFloatType

Reply:
Oh, I think `HalfFloatType` doesn't have CType defined: https://github.com/apache/arrow/blob/5e1a4fd8a4ed3630c9549c611222d2d6c32357ca/cpp/src/arrow/type_traits.h#L302-L314

Reply:
Ah, yes, I was confusing the `TypeTraits<>::CType` with `HalfFloatType::c_type` (and the latter does exist, I think. It's also a bit confusing this is not consistent, probably something that could be changed)